hail/src/main/scala/is/hail/backend/service/Worker.scala

```diff
@@ -89,18 +101,21 @@ object Worker {
           context
         }
       }
-    }(fileRetrievalExecutionContext)
+    }
 
-    // retryTransientErrors handles timeout and exception throwing logic
     val f = Await.result(fFuture, Duration.Inf)
     val context = Await.result(contextFuture, Duration.Inf)
 
     timer.end("readInputs")
     timer.start("executeFunction")
 
-    val hailContext = HailContext(
-      // FIXME: workers should not have backends, but some things do need hail contexts
-      new ServiceBackend(null), skipLoggingConfiguration = true, quiet = true)
+    if (HailContext.isInitialized) {
+      HailContext.get.backend = new ServiceBackend(null, null, null)
+    } else {
+      HailContext(
+        // FIXME: workers should not have backends, but some things do need hail contexts
```

Comment:
worker code shouldn't need this, I think. Let's follow up when the PR goes in to remove this.

Reply:
OK. I've added to my todo.